src/openforms/contrib/zgw/api/views.py

```diff
@@ -75,22 +75,51 @@ class BaseInformatieObjectTypenListView(ListMixin[DocumentType], APIView):
 
     authentication_classes = (authentication.SessionAuthentication,)
     permission_classes = (permissions.IsAdminUser,)
-    serializer_class = InformatieObjectTypeSerializer
+    serializer_class = DocumentTypeSerializer
     filter_serializer_class: ClassVar[type[DocumentTypesFilter]]
 
     def get_objects(self) -> list[DocumentType]:
         filter_serializer = self.filter_serializer_class(data=self.request.query_params)
         filter_serializer.is_valid(raise_exception=True)
 
-        catalogus_url = filter_serializer.validated_data["catalogus_url"]
+        catalogue_url: str = filter_serializer.validated_data["catalogue_url"]
+        case_type_identification: str = filter_serializer.validated_data[
+            "case_type_identification"
+        ]
 
         document_types: list[DocumentType] = []
         with filter_serializer.get_ztc_client() as client:
+            # if a case type identification filter is provided, filter down the list
+            # of document types to the ones present in any version of the case type
```

Comment:
Should it be narrowed down to only the active versions?

Reply:
no because that breaks when you prepare things for future versions. Either way, there is potential for runtime crashes, so I opted to go as wide as possible and rely on the digest email to report resolution issues.